omod/src/main/java/org/openmrs/module/pihcore/RequestMonitoringFilter.java

```diff
@@ -76,53 +44,29 @@ public void doFilter(ServletRequest request, ServletResponse response, FilterCha
 		chain.doFilter(request, response);
 		long endTime = System.currentTimeMillis();
 		long totalTime = endTime - startTime;
-		if (enabled && request instanceof HttpServletRequest) {
-			try {
-				HttpServletRequest httpRequest = (HttpServletRequest) request;
-				HttpSession session = httpRequest.getSession();
-				ISO8601DateFormat dateFormat = new ISO8601DateFormat();
-				SimpleObject logLine = new SimpleObject();
-				logLine.put("timestamp", dateFormat.format(new Date(startTime)));
-				logLine.put("sessionId", session.getId());
-				logLine.put("user", session.getAttribute("username"));
-				logLine.put("loadTime", totalTime);
-				logLine.put("method", httpRequest.getMethod());
-				logLine.put("requestPath", httpRequest.getRequestURI());
-				logLine.put("queryParams", formatRequestParams(httpRequest));
-				logger.info(logLine.toJson());
-			}
-			catch (Exception e) {
-				// Do nothing, we don't want this filter to cause any adverse behavior
+		if (logger.isTraceEnabled()) {
+			if (request instanceof HttpServletRequest) {
+				try {
+					HttpServletRequest httpRequest = (HttpServletRequest) request;
+					HttpSession session = httpRequest.getSession();
+					ThreadContext.put("sessionId", session.getId());
+					ThreadContext.put("user", (String) session.getAttribute("username"));
+					ThreadContext.put("loadTime", Long.toString(totalTime));
+					ThreadContext.put("method", httpRequest.getMethod());
+					ThreadContext.put("ipAddress", httpRequest.getRemoteAddr());
+					ThreadContext.put("servletPath", httpRequest.getServletPath());
+					ThreadContext.put("requestURI", httpRequest.getRequestURI());
```

Comment:
How does this thread context stuff work? does it get logged by the next "logger.trace" line?

Reply:
It is available to log - it comes down to what you specify in your layout.  Any item in the ThreadContext can be included in the logging pattern / referenced by the particular appender as:  %X{key}